### DISCUSSION TITLE
How to validate form item array ?

### DISCUSSION THREAD

**Discussion by palamike** [tusen-ai/naive-ui#1371] [Q&A]
Hi,

I have form contains array of fields.  for example..

I would have data structure like this.

```
order = { 
   id: 'id of order',
  date: 'date of order',
  items: [ // order items
   { item_name: 'item 1 name', price: 'item 1 price'  },
   { item_name: 'item 2 name', price: 'item 2 price'  }
  ]
}
```

How can i validate items[i].item_name and items[i].price ? any suggestion ?

**Comment by 07akioni** [comment]
Set path for form-item. For example if you want to validate items, set path to `items`

**Reply by palamike** [reply]
For recorded..

I did it by set path like items.a01.item_name, items.a02.item_name to work on each fields. 
If the array is dynamic rules must be dynamic also.

**Reply by 07akioni** [reply]
I'm not sure what's your scene, but it's possible to set rule on each form item. If you can pass the rule in each form item, it would be dynamic since the rule is added and removed along with the rendered form item.
